tstest/integration/vms/vms_test.go

```diff
@@ -729,35 +731,48 @@ func testDistro(t *testing.T, loginServer string, d Distro, signer ssh.Signer, i
 
 	t.Log("opened session")
 
-	_, _, err = e.Expect(regexp.MustCompile(`(\#)`), timeout)
-	if err != nil {
-		t.Fatalf("%d: can't get a shell: %v", port, err)
+	var batch = []expect.Batcher{
+		&expect.BSnd{S: "PS1='# '\n"},
+		&expect.BExp{R: `(\#)`},
 	}
-	t.Logf("got shell for %d", port)
+
 	switch d.initSystem {
 	case "openrc":
 		// NOTE(Xe): this is a sin, however openrc doesn't really have the concept
 		// of service readiness. If this sleep is removed then tailscale will not be
 		// ready once the `tailscale up` command is sent. This is not ideal, but I
 		// am not really sure there is a good way around this without a delay of
 		// some kind.
-		err = e.Send("rc-service tailscaled start && sleep 2\n")
+		batch = append(batch, &expect.BSnd{S: "rc-service tailscaled start && sleep 2\n"})
 	case "systemd":
-		err = e.Send("systemctl start tailscaled.service\n")
-	}
-	if err != nil {
-		t.Fatalf("can't send command to start tailscaled: %v", err)
-	}
-	_, _, err = e.Expect(regexp.MustCompile(`(\#)`), timeout)
-	if err != nil {
-		t.Fatalf("%d: can't get a shell: %v", port, err)
-	}
-	err = e.Send(fmt.Sprintf("tailscale up --login-server %s\n", loginServer))
-	if err != nil {
-		t.Fatalf("%d: can't send tailscale up command: %v", port, err)
-	}
-	_, _, err = e.Expect(regexp.MustCompile(`Success.`), timeout)
+		batch = append(batch, &expect.BSnd{S: "systemctl start tailscaled.service\n"})
+	}
+
+	batch = append(batch,
+		&expect.BExp{R: `(\#)`},
```

Comment:
You can remove some boilerplate here with:

```
batch = append(batch, []*expect.BExp{
        {R: `(\#)`},
        {S: `...`},
}...)
```

Reply:
They are different types for sending and receiving.

Reply:
Oh so they are. I misread it as using different fields on the same type.